synapse/handlers/federation_event.py

```diff
@@ -1023,9 +1023,15 @@ async def _handle_marker_event(self, origin: str, marker_event: EventBase):
             return
 
         # Skip processing a marker event if the room version doesn't
-        # support it.
+        # support it or the event is not from the room creator.
         room_version = await self.store.get_room_version(marker_event.room_id)
-        if not room_version.msc2716_historical:
+        create_event = await self.store.get_create_event_for_room(marker_event.room_id)
+        room_creator = create_event.content.get(EventContentFields.ROOM_CREATOR)
```

Comment:
@MadLittleMods sorry to raise this PR from the dead, but can you remember if there is a reason we use `room_creator = create_event.content.get(EventContentFields.ROOM_CREATOR)` here, but select the `creator` column from `rooms` elsewhere in the same PR?

Reply:
Probably an oversight. I can't think of a reason why it would matter. We just want the room creator which doesn't change.

For the `insertion` and `batch` events, we switched from the `m.room.create` event lookup and moved to the `rooms` table lookup in https://github.com/matrix-org/synapse/pull/10566/commits/aafa06981654b412d6d92b786f83129cab360303

@richvdh Are you touching this in a different refactor to update this?